### PR TITLE
Support for audio only streaming

### PR DESCRIPTION
**Pull request by dave-malone**
*Issue #, if available:*
n/a

*Description of changes:*

Adding support for audio only streaming, including support for createRealtimeAudioStreamInfoProviderWithCodecs and createOfflineRealtimeAudioStreamInfoProviderWithCodecs. Changed createAudioTrackInfo api to require the audio track ID to be specified as an argument instead of hard coding the DEFAULT_AUDIO_TRACK_ID within the function implementation. This allows calling APIs to add audio tracks with different track IDs. Finally, added new kvsAudioOnlyStreamingSample application to demonstrate how to transmit audio only.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/KvsAudioOnlyStreamingSample.c [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    PDeviceInfo pDeviceInfo = NULL;
+    PStreamInfo pStreamInfo = NULL;
+    PClientCallbacks pClientCallbacks = NULL;
+    PStreamCallbacks pStreamCallbacks = NULL;
+    CLIENT_HANDLE clientHandle = INVALID_CLIENT_HANDLE_VALUE;
+    STREAM_HANDLE streamHandle = INVALID_STREAM_HANDLE_VALUE;
+    STATUS retStatus = STATUS_SUCCESS;
+    PCHAR accessKey = NULL, secretKey = NULL, sessionToken = NULL, streamName = NULL, region = NULL, cacertPath = NULL;
+    UINT64 streamStopTime, streamingDuration = DEFAULT_STREAM_DURATION, fileSize = 0;
+    TID audioSendTid, videoSendTid;
```
Comment (hassanctech) [inline samples/KvsAudioOnlyStreamingSample.c:104]: Let's remove all video related variables to clean up this sample so `videoSendTid` which is the thread id for that routine can be removed.

File: samples/KvsAudioOnlyStreamingSample.c [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    CHK_STATUS(createStreamCallbacks(&pStreamCallbacks));
+    CHK_STATUS(addStreamCallbacks(pClientCallbacks, pStreamCallbacks));
+
+    CHK_STATUS(createKinesisVideoClient(pDeviceInfo, pClientCallbacks, &clientHandle));
+    CHK_STATUS(createKinesisVideoStreamSync(clientHandle, pStreamInfo, &streamHandle));
+
+    data.streamStopTime = streamStopTime;
+    data.streamHandle = streamHandle;
+    data.streamStartTime = defaultGetTime();
+    ATOMIC_STORE_BOOL(&data.firstVideoFramePut, FALSE);
```
Comment (hassanctech) [inline samples/KvsAudioOnlyStreamingSample.c:237]: To better match the context can we change this variable name to `firstAudioFramePut`

File: src/source/StreamInfoProvider.h [hunk 3]
```diff
@@ -15,6 +15,7 @@ extern "C" {
 
 #define VIDEO_ONLY_TRACK_COUNT       1
 #define VIDEO_WITH_AUDIO_TRACK_COUNT 2
+#define AUDIO_ONLY_TRACK_COUNT 1
```
Comment (hassanctech) [inline src/source/StreamInfoProvider.h:18]: Can we add spaces here to align with the others

File: src/source/StreamInfoProvider.c [hunk 4]
```diff
@@ -103,11 +115,12 @@ STATUS createVideoTrackInfo(VIDEO_CODEC_ID videoCodecId, PCHAR contentType, PTra
 }
 
 // Creates track info for audio with given codec and sets the content type as per the codec
-STATUS createAudioTrackInfo(AUDIO_CODEC_ID audioCodecId, PCHAR contentType, PTrackInfo pTrackInfo)
+STATUS createAudioTrackInfo(AUDIO_CODEC_ID audioCodecId, PCHAR contentType, PTrackInfo pTrackInfo, UINT32 audioTrackId)
 {
     STATUS retStatus = STATUS_SUCCESS;
 
-    pTrackInfo->trackId = DEFAULT_AUDIO_TRACK_ID;
+    // pTrackInfo->trackId = pTrackInfo->trackId + 1;
```
Comment (hassanctech) [inline src/source/StreamInfoProvider.c:122]: remove commented line

File: samples/KvsAudioOnlyStreamingSample.c [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    PTrackInfo pAudioTrack = NULL;
+    CHAR audioCodec[AUDIO_CODEC_NAME_MAX_LENGTH];
+    BYTE aacAudioCpd[KVS_AAC_CPD_SIZE_BYTE];
+    BYTE alawAudioCpd[KVS_PCM_CPD_SIZE_BYTE];
+
+    MEMSET(&data, 0x00, SIZEOF(SampleCustomData));
+
+    STRNCPY(audioCodec, AUDIO_CODEC_NAME_AAC, STRLEN(AUDIO_CODEC_NAME_AAC)); //aac audio by default
+
+    // if (argc == 6) {
```
Comment (hassanctech) [inline samples/KvsAudioOnlyStreamingSample.c]: delete commented out lines

File: samples/KvsAudioOnlyStreamingSample.c (hunk 2 above)
Comment (dave-malone) [reply samples/KvsAudioOnlyStreamingSample.c]: Removed this variable altogether since it is not being used in the audio only sample program

**Pull request merged**